src/components/useVerboseFormik.ts

```diff
@@ -0,0 +1,97 @@
+// Software, and to permit persons to whom the Software is furnished to do so,
+// subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
+// INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A
+// PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+// HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+// OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
+// SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+//
+
+import { FormikValues, useFormik } from "formik";
+import {
+  FormFieldStates,
+  useFieldStates,
+  VerboseFieldState,
+} from "../lib/formValidationTypes";
+
+/**
+ * Extends the {@link useFormik} function by adding state property supporting
+ * verbose field state management.
+ *
+ * @param initialValues The initial values for the form fields.
+ * @param initialStates The initial states of the form fields.
+ *
+ * @returns formik object
+ */
+export default function useVerboseFormik<FormParameters extends FormikValues>({
+  initialValues,
+  initialStates,
+}: {
+  initialValues: FormParameters;
+  initialStates: FormFieldStates<FormParameters, VerboseFieldState | undefined>;
+}) {
+  const states = useFieldStates<FormParameters, VerboseFieldState | undefined>(
+    initialStates
+  );
+
+  const formik = useFormik<FormParameters>({
+    initialValues,
+    onSubmit: () => {},
+    validateOnBlur: false,
+    validateOnChange: false,
+  });
+
+  const modifiedFormik: Omit<
+    typeof formik,
+    "errors" | "setErrors" | "initialErrors" | "setFieldErrors"
+  > & {
+    states: typeof states;
+  } = { ...formik, states };
+
+  // Helper to add an array field's child value.
+  const addChild = async (fieldName: keyof FormParameters, value: unknown) => {
+    await formik.setValues({
+      ...formik.values,
+      [fieldName]: [...formik.values[fieldName], value],
+    });
+    states.setChildren(fieldName, [
+      ...(states.all[fieldName].childStates || []),
+      undefined,
+    ]);
+  };
+
+  // Helper to remove an array field's child value.
+  const removeChild = async (
+    fieldName: keyof FormParameters,
+    index: number
+  ) => {
+    const values = formik.values[fieldName];
+    if (!Array.isArray(values)) {
+      throw new Error(
+        "Cannot remove child form value from an object that's not an array."
+      );
+    }
+    await formik.setValues({
+      ...formik.values,
+      [fieldName]: values.filter((_: unknown, i: number) => i !== index),
+    });
+    states.setChildren(
+      fieldName,
+      states.all[fieldName].childStates?.filter((_, i) => i !== index) || []
+    );
+  };
+
+  return { ...modifiedFormik, addChild, removeChild };
```

Comment:
you'll want to `useMemo` here.

Reply:
I'm not quite sure how I would employ this here and I couldn't quite make sense of the documentation I found.

Wrapping the whole function makes the linter say: `React Hook "useFieldStates" cannot be called inside a callback. React Hooks must be called in a React function component or a custom React Hook function.`
If only wrapping parts, the linter comoplains: `React Hook useMemo has missing dependencies`

Could you give me a hint here?